### ISSUE TITLE
Privileges on merge table not propagated to partition tables

### ISSUE THREAD

**Issue by bartscheers** [MonetDB/MonetDB#7402]
**Describe the bug**

Granting select privileges to a user on merge table do not propagated properly to partition table. When set on the merge table a query with where clause on the merge table fails unexpectedly.

**To Reproduce**
Create a setting with minimal input for an external user to demonstrate the buggy behavior.
This includes the relevant part of the database schema description.
Performance trace of the rogue query (using the TRACE command)

1. Start the daemon and server and create the database instance using the `setup-db.sh` script.
2. Run the `run-test.sh` script to setup the schema, create the merge and partition tables, create the user and select privilege, run the allowed queries (of which some fail due to the bug) and denied queries (expected to fail, because of insufficient privileges). The run script also drops a partition table and recreates it again, after which the allowed queries are executed again, to see if they succeed, but they don't.

`setup-db.sh`:
```
monetdb status
monetdb stop bugdb
monetdb destroy -f bugdb
monetdb create bugdb
monetdb start bugdb
monetdb release bugdb
monetdb status
```

`run-test.sh` (note the different credentials):
```
DOTMONETDBFILE=~/.monetdb mclient -d bugdb < setup-schema.sql
DOTMONETDBFILE=~/.adam mclient -d bugdb < setup-tables.sql
DOTMONETDBFILE=~/.adam mclient -d bugdb < setup-user-grants.sql

echo -e "\n..:: This block of queries should *ALL* succeed because of permission allowed. However, ... ::.."
DOTMONETDBFILE=~/.user1 mclient -d bugdb < select-with-priv-allowed.sql
echo -e "..:: End of block of permission allowed queries ::.."

echo -e "\n..:: The next block of queries should *ALL* fail because of permission denied ::.."
DOTMONETDBFILE=~/.user1 mclient -d bugdb < select-with-priv-denied.sql
echo -e "..:: End of block of permission denied queries ::.."

DOTMONETDBFILE=~/.adam mclient -d bugdb < setup-drop-recreate-table.sql
DOTMONETDBFILE=~/.user1 mclient -d bugdb < select-with-priv-allowed.sql
```

`setup-schema.sql`, `setup-tables.sql`, `setup-user-grants.sql`:
```
ALTER USER "monetdb" RENAME TO "adam";
ALTER USER SET PASSWORD 'adam123' USING OLD PASSWORD 'monetdb';
CREATE SCHEMA "myschema" AUTHORIZATION "adam";
ALTER USER "adam" SET SCHEMA "myschema";

create merge table mt (i int, d double);
create table t1 (i int, d double);
create table t2 (i int, d double);
create table t3 (i int, d double);
alter table mt add table t1;
alter table mt add table t2;
alter table mt add table t3;
insert into t1 values (1, pi()),(2, exp(1));
insert into t2 values (10, 300),(20, 400);
insert into t3 values (100, 100.0),(200, 200.01);
create table u1 (i int, d double, t text);
insert into u1 values (1000, 2.3456, 'seven');
create table v1 (i int, j int);
insert into v1 values (10000, 1234);

create user user1 with unencrypted password 'user123' name 'User1' schema "myschema";
grant select on table "myschema".mt to "user1";
grant select on table "myschema".u1 to "user1";
```

`select-with-priv-allowed.sql`, `select-with-priv-denied.sql`:
```
-- These should be allowed. However, query 2-6 fail.
select count(*) from mt;
select count(*) from mt where i = 1;
select * from mt;
select * from mt where i = 1;
select * from mt where i = 10;
select * from mt where i = 100;
select * from u1;

-- These should be denied
select * from t1;
select * from t1 where i = 1;
select * from t2;
select * from t2 where i = 10;
select * from t3;
select * from t3 where i = 100;
select * from v1;
```

`setup-drop-recreate-table.sql`:
```
alter table mt drop table t1;
drop table t1;
create table t1 (i int, d double);
alter table mt add table t1;
insert into t1 values (1, 3.1415), (2, 2.7182), (3, 5.8597);
```

After this the `select-with-priv-allowed.sql` queries should succeed again, but they don't.

**Expected behavior**

The `run-test.sh` script has allowed queries which should all succeed

**Screenshots**

**Software versions**
 - MonetDB version number [a milestone label]
 - OS and version: [e.g. Ubuntu 18.04]
 - Installed from release package or self-installed and compiled

MonetDB version Jun2023-SP1
Fedora 37 (6.4.13)
Installed using cmake

**Issue labeling **
Make liberal use of the labels to characterise the issue topics. e.g. identify severity, version, etc..

**Additional context**

Scripts and queries are attached.

**Comment by smavros** [comment]
@bartscheers the `GRANT obj_privs ON merge_table ...` should not propagate any privileges to the child tables. If user _user1_ is granted `SELECT` on _mt_ then they should not have `SELECT` on _t1_, _t2_ or _t3_, exactly as you expect in `select-with-priv-denied.sql` examples. The reason why `select-with-priv-allowed.sql` queries 2-6 fail is due to a bug. We erroneously check the privileges of not only the merge table but the child tables. The child table privs are irrelevant. 

This is already fixed with 07ca69cefb06f2cb3313949643fc1b33d40f537c on default and will be included in the next release.

**Comment by bartscheers** [comment]
Thanks Stefanos, all clear. I am looking forward to the next release ;)
